src/Recorders/Concerns/Thresholds.php

```diff
@@ -2,13 +2,36 @@
 
 namespace Laravel\Pulse\Recorders\Concerns;
 
+use Illuminate\Support\Facades\Config;
+
 trait Thresholds
 {
     /**
      * Determine if the duration is under the configured threshold.
      */
-    protected function underThreshold(int|float $duration): bool
+    protected function underThreshold(int|float $duration, string $value): bool
+    {
+        return $duration < $this->threshold($value);
+    }
+
+    /**
+     * Get the threshold for the given value.
+     */
+    protected function threshold(string $value, ?string $recorder = null): int
     {
-        return $duration < $this->config->get('pulse.recorders.'.static::class.'.threshold');
+        $recorder ??= static::class;
+
+        $config = Config::get("pulse.recorders.{$recorder}.threshold");
+
+        if (! is_array($config)) {
+            return $config;
+        }
+
+        // @phpstan-ignore argument.templateType, argument.templateType
+        $custom = collect($config)
+            ->except(['default'])
+            ->first(fn ($threshold, $pattern) => preg_match($pattern, $value));
+
+        return $custom ?? $config['default'] ?? 1_000;
```

Comment:
Notice that we have a hard-coded `1_000ms` here as the default threshold when no default is specified; this matches our config defaults.

When a user changes from a single threshold to an array, if they forget the `default` key that the app continues to work with this default.

Not sure if we wanna keep this or just document that they need to have the `default` key, but this felt okay to me.

Reply:
I'm not sure if you want my opinion about this, but don't breaking the app even without a `default` key sounds good for me. Although, if you want help documenting the `default` key's mandatory, count on me.